crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -1064,16 +1064,22 @@ impl<P: RoomDataProvider> TimelineController<P> {
 
                     match decryptor.decrypt_event_impl(original_json).await {
                         Ok(event) => {
-                            trace!(
-                                "Successfully decrypted event that previously failed to decrypt"
-                            );
-
-                            // Notify observers that we managed to eventually decrypt an event.
-                            if let Some(hook) = unable_to_decrypt_hook {
-                                hook.on_late_decrypt(&remote_event.event_id, *utd_cause).await;
+                            if let matrix_sdk::deserialized_responses::TimelineEventKind::UnableToDecrypt { utd_info, ..} = event.kind {
+                                info!("Failed to decrypt event after receiving room key: {:?}", utd_info.reason);
+                                None
+                            } else {
+                                trace!(
+                                    kind = ?event.kind,
```

Comment:
this is probably a bit verbose?

Reply:
Not changed by this PR. Any other reason to remive it? is it spamming the RS?

Reply:
Contains most of the event's data that's not privacy-sensitive, so it will be spammy. This line can be reached multiple times for each undecrypted event that failed to re-decrypt again, so it will be super spammy, please remove it.